src/repository.c

```diff
@@ -2211,6 +2212,59 @@ Repository_set_refdb(Repository *self, Refdb *refdb)
     Py_RETURN_NONE;
 }
 
+static int foreach_stash_cb(size_t index, const char *message, const git_oid *stash_id, void *payload)
+{
+    int err;
+    Stash *py_stash;
+
+    py_stash = PyObject_New(Stash, &StashType);
+    if (py_stash == NULL)
+        return GIT_EUSER;
+
+    py_stash->oid = git_oid_to_python(stash_id);
+    py_stash->message = (message != NULL) ? strdup(message) : NULL;
```

Comment:
As said in the other comments, I think `message` and `stash_id` cannot be `NULL`. So please add an `assert` statement for each one. Then check whether `git_oid_to_python` or `strdup` return `NULL`, that would mean a memory error.